src/main/java/net/juniper/netconf/element/Hello.java

```diff
@@ -0,0 +1,165 @@
+     */
+    public static Hello from(final String xml)
+            throws ParserConfigurationException, IOException, SAXException, XPathExpressionException {
+
+        final DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
+        documentBuilderFactory.setNamespaceAware(true);
+        final Document document = documentBuilderFactory.newDocumentBuilder()
+                .parse(new InputSource(new StringReader(xml)));
+        final XPath xPath = XPathFactory.newInstance().newXPath();
+        final String sessionId = xPath.evaluate("/*[namespace-uri()='urn:ietf:params:xml:ns:netconf:base:1.0' and local-name()='hello']/*[namespace-uri()='urn:ietf:params:xml:ns:netconf:base:1.0' and local-name()='session-id']", document);
+        final HelloBuilder builder = Hello.builder()
+                .originalDocument(document)
+                .sessionId(sessionId);
+        final NodeList capabilities = (NodeList) xPath.evaluate("/*[namespace-uri()='urn:ietf:params:xml:ns:netconf:base:1.0' and local-name()='hello']/*[namespace-uri()='urn:ietf:params:xml:ns:netconf:base:1.0' and local-name()='capabilities']/*[namespace-uri()='urn:ietf:params:xml:ns:netconf:base:1.0' and local-name()='capability']", document, XPathConstants.NODESET);
+        for (int i = 0; i < capabilities.getLength(); i++) {
+            final Node node = capabilities.item(i);
+            builder.capability(node.getTextContent());
+        }
+        final Hello hello = builder.build();
+        if (log.isInfoEnabled()) {
+            log.info("hello is: {}", hello.toXML());
+        }
+        return hello;
+    }
+
+    @Builder
+    private Hello(
+            final Document originalDocument,
+            final String namespacePrefix,
+            final String sessionId,
+            @Singular("capability") final List<String> capabilities) {
+        this.sessionId = sessionId;
+        this.capabilities = capabilities;
+        if (originalDocument != null) {
+            this.document = originalDocument;
+        } else {
+            this.document = createDocument(namespacePrefix, sessionId, capabilities);
+        }
+        this.xml = createXml(document);
+    }
+
+    private static Document createDocument(
+            final String namespacePrefix,
+            final String sessionId,
+            final List<String> capabilities) {
+
+        final DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
+        documentBuilderFactory.setNamespaceAware(true);
+        final Document createdDocument;
+        try {
+            createdDocument = documentBuilderFactory.newDocumentBuilder().newDocument();
+        } catch (final ParserConfigurationException e) {
+            throw new IllegalStateException("Unable to create document builder", e);
+        }
+
+        final Element helloElement
+                = createdDocument.createElementNS(NetconfConstants.URN_XML_NS_NETCONF_BASE_1_0, "hello");
```

Comment:
When I see four lines that are very similar, I wonder if it could be turned into a private function. it looks like the difference is the name space name? So if you could pass in a Document and name space name and it would return an Element?

I'd also be tempted to make the name space name an enum and use that instead of the string.